src/ansi-c/c_typecheck_type.cpp

```diff
@@ -560,10 +557,15 @@ void c_typecheck_baset::typecheck_array_type(array_typet &type)
     throw 0;
   }
 
-  // check size, if any
+  // We add the index type.  This is added as a comment now for backwards
+  // compatibility, but will eventually be the first subtype.
```

Comment:
I'm worried that this comment will eventually (hopefully) be inconsistent with reality and no one will notice. I think this comment should only go in the comments of `array_typet`, but not in any place that (correctly) uses the API.

Reply:
Ok, will move.